definitions/reports/advisor_on_prem_remediations.rb

```diff
@@ -0,0 +1,27 @@
+module Reports
+  class AdvisorOnPremRemediations < ForemanMaintain::Report
+    metadata do
+      description 'Report number of remediations based on advisor rules from advisor on premise'
+      confine do
+        feature(:installer)&.answers&.dig('foreman::plugin::rh_cloud', 'enable_iop_advisor_engine')
+      end
+    end
+
+    def run
+      data_field('advisor_on_prem_remediations') { advisor_on_prem_remediations }
+    end
+
+    def advisor_on_prem_remediations
+      sql_count('(
+        SELECT DISTINCT hits.id
+        FROM job_invocations AS jobs
+        INNER JOIN remote_execution_features AS rexf ON jobs.remote_execution_feature_id = rexf.id
+        INNER JOIN template_invocations AS tinv ON jobs.id = tinv.job_invocation_id
+        INNER JOIN insights_hits AS hits ON tinv.host_id = hits.host_id
+        INNER JOIN foreman_tasks_tasks AS tasks ON tinv.run_host_job_task_id = tasks.id
+        WHERE rexf.label = \'rh_cloud_remediate_hosts\'
+        AND tasks.result = \'success\'
+      ) as subquery')
```

Comment:
The parens and `as subquery` is unnecessary

Reply:
It doesn't work without it, `report generate` does not return a value and trying to replicate the query on db console throws an error.
```
foreman=# SELECT COUNT(*) FROM (SELECT DISTINCT tinv.host_id, jobs.id
        FROM job_invocations AS jobs
        INNER JOIN remote_execution_features AS rexf ON jobs.remote_execution_feature_id = rexf.id
        INNER JOIN template_invocations AS tinv ON jobs.id = tinv.job_invocation_id
        WHERE rexf.label = 'rh_cloud_remediate_hosts'
        AND tinv.host_id IS NOT NULL);
ERROR:  subquery in FROM must have an alias
LINE 1: SELECT COUNT(*) FROM (SELECT DISTINCT tinv.host_id, jobs.id
                             ^
HINT:  For example, FROM (SELECT ...) [AS] foo.
foreman=# 
```

Reply:
Eh, right, but we don't really need a distinct in there, do we? job invocation has many template invocations, template invocation belongs to a single host, but a single host can't be in a single job invocation multiple times.

Reply:
> I understood this as counting each host in a single job invocation only once

Well, yes. But a host either is or is not in a job, a single host can't be in a single job multiple times. We don't need to do anything to remove duplicates as there are no duplicates.